validateOffload.py

```diff
@@ -10,14 +12,83 @@
 from task import Task
 from testConfig import TestConfig
 from tftbase import PluginOutput
+from tftbase import PluginResult
 from tftbase import PodType
 from tftbase import TFT_TOOLS_IMG
+from tftbase import TestMetadata
 from tftbase import TftAggregateOutput
-from tftbase import VALIDATE_OFFLOAD_PLUGIN
 from thread import ReturnValueThread
 
+VF_REP_TRAFFIC_THRESHOLD = 1000
+
+
+def no_traffic_on_vf_rep(
+    rx_start: int, tx_start: int, rx_end: int, tx_end: int
+) -> bool:
+    return (
+        rx_end - rx_start < VF_REP_TRAFFIC_THRESHOLD
+        and tx_end - tx_start < VF_REP_TRAFFIC_THRESHOLD
+    )
+
+
+class PluginValidateOffload(pluginbase.Plugin):
+    PLUGIN_NAME = "validate_offload"
+
+    def enable(
+        self,
+        *,
+        tc: TestConfig,
+        node_server_name: str,
+        node_client_name: str,
+        perf_server: perf.PerfServer,
+        perf_client: perf.PerfClient,
+        tenant: bool,
+    ) -> list[Task]:
+        # TODO allow this to run on each individual server + client pairs.
+        return [
+            ValidateOffload(tc, perf_server, tenant),
+            ValidateOffload(tc, perf_client, tenant),
+        ]
+
+    def eval_log(
```

Comment:
I like the use of the abstract class to add plugin.enable.

I am not so sure about moving the evaluation logic into the plugin. Originally, the goal was to have this code separated into two parts:

- the flow test tool which takes in a config and creates a raw json output
- the evaluator which takes in that json output, parses out the meaningful data, and determines success/failure.

It seems like these roles would be less differentiated by moving the evaluation logic into the plugin class. WDYT?

Reply:
for now, there is only one plugin-specific implementation of `eval_log()`. It's hard to guess, whether a new plugin will have widely different API, that makes it hard to abstract. In any case, when that plugin gets added, then it's easy to modify the code, so that it can again be good. Maybe something like:

```
    plugin = pluginbase.get_by_name(plugin_output.name)
    if plugin.eval_log_is_special_kind():
          plugin_result = plugin.eval_log_special_kind(special_args...)
    else:
          plugin_result = plugin.eval_log(plugin_output, run["flow_test"].tft_metadata)
```

in any case, for the currently existing code, this abstract well.

---

> the evaluator which takes in that json output, parses out the meaningful data, and determines success/failure.

This didn't change, did it?

All that the branch does at this place is replace
```python
        if plugin_output.name == VALIDATE_OFFLOAD_PLUGIN:
                    self._eval_validate_offload(...
```
with
```python
                plugin = pluginbase.get_by_name(plugin_output.name)
                plugin_result = plugin.eval_log(
                     plugin_output, run["flow_test"].tft_metadata
                 )
```

It really just moves code around. And having code specific to "validate_offload" plugin in `evaluate.py` is not nice.

Reply:
Yeah, make sense to me